crates/metrics/src/buckets.rs

```diff
@@ -0,0 +1,141 @@
+use std::{
+    collections::HashMap,
+    sync::OnceLock,
+};
+#[cfg(test)]
+use strum_macros::EnumIter;
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(test, derive(EnumIter))]
+pub(crate) enum Buckets {
+    Timing,
+    GasUsed,
+    TransactionsCount,
+    Fee,
+    SizeUsed,
+}
+static BUCKETS: OnceLock<HashMap<Buckets, Vec<f64>>> = OnceLock::new();
+pub(crate) fn buckets(b: Buckets) -> impl Iterator<Item = f64> {
+    BUCKETS.get_or_init(initialize_buckets)[&b].iter().copied()
+}
+
+#[rustfmt::skip]
+fn initialize_buckets() -> HashMap<Buckets, Vec<f64>> {
+    [
+        (
+            Buckets::Timing,
+            vec![
+                0.005,
+                0.010,
+                0.025,
+                0.050,
+                0.100,
+                0.250,
+                0.500,
+                1.000,
+                2.500,
+                5.000,
+               10.000,
+            ],
+        ),
+        (
+            // Gas limit is 30_000_000.
+            Buckets::GasUsed,
+            vec![
+                    10_000.0,
+                    25_000.0,
+                    50_000.0,
+                   100_000.0,
+                   250_000.0,
+                   500_000.0,
+                 1_000_000.0,
+                 3_000_000.0,
+                 7_500_000.0,
+                15_000_000.0,
+                25_000_000.0,
+            ],
+        ),
+        (
+            // Transaction count is fixed at 65_535.
+            Buckets::TransactionsCount,
```

Comment:
The low end could also be informative in metrics. 1 is good to know and it's a very state of the chain between 1 and even 2 or 3, so I'd prefer to have 1-5 all there I think. And then probably 200, 400, 800, 1600, 3200 on the high end for now. We can always add more.

Reply:
I updated these buckets so that we are very detailed at low, then we reach your proposed high end around 16th bucket, but then I left a few at the very end for the high values. Commit is here: https://github.com/FuelLabs/fuel-core/pull/2310/commits/ef13002a49b4f7eba35010f0be8d3d6c4638d770

Distribution visualisation:
![image](https://github.com/user-attachments/assets/7fee8a03-f812-4ea7-a15f-263213d3c247)